parfive/downloader.py

```diff
@@ -372,26 +375,120 @@ def callback(token, future, main_pb):
                                           total=get_http_size(resp))
                     else:
                         file_pb = None
-                    with open(str(filepath), 'wb') as fd:
-                        while True:
-                            chunk = await resp.content.read(chunksize)
-                            if not chunk:
-                                # Close the file progressbar
-                                if file_pb is not None:
-                                    file_pb.close()
+                    if max_splits and resp.headers.get('Accept-Ranges', None) == "bytes":
+                        # XXX: int(?)
+                        content_length = int(resp.headers['Content-length'])
+                        split_length = content_length//max_splits
+                        ranges = [
+                            [start, start + split_length]
+                            for start in range(0, content_length, split_length)
+                        ]
+                        # let the last part download everything
+                        ranges[-1][1] = ''
+                        queue = asyncio.Queue()
+                        workers = []
+                        for _range in ranges:
+                            workers.append(
+                                asyncio.create_task(Downloader._ranged_http_producer(
```

Comment:
If you are calling other methods from here you should make this a regular method.